fuel-core/src/model/fuel_block.rs

```diff
@@ -69,72 +67,84 @@ impl BlockHeight {
     }
 }
 
-#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
-pub struct TransactionCommitment {
-    pub sum: Word,
-    pub root: Bytes32,
-}
-
 #[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct FuelBlockHeaders {
-    pub fuel_height: BlockHeight,
+pub struct FuelBlockHeader {
+    /// Fuel block height.
+    pub height: BlockHeight,
+    /// Ethereum block number.
+    pub number: BlockHeight,
```

Comment:
This is not something we know at moment of block creation, and we cant predict at what exact eth block is fuel block going to be commited. But it makes sense to be included as a field as it is going to be updated when fuel blocks get commited over eth. I would put this as `Option` or maybe zeroed value, and it needs to be removed from `id` hash calculation.

Reply:
It seems hard to avoid non-deterministic behavior between validators if we don't commit to the ethereum global state as of this block. The block producer should be able to pick an ethereum block to claim that they are including all the ethereum events up to that point.

cc @adlerjohn

Reply:
I was thinking this would be L1 block number used for synchronizing deposits, not the L1 block that this block ended up being published in.

Reply:
That's a possibility, yes!

Reply:
I'm okay with leaving a field in the header for Ethereum block, but there should be a description here that there's some caveats, and linking to https://github.com/FuelLabs/fuel-specs/issues/220

Reply:
updated the description of this field, lmk if that works for you.

Reply:
Good enough!